frontend/javascripts/oxalis/controller/combinations/volumetracing_plane_controller.js

```diff
@@ -70,17 +71,21 @@ const simulateTracing = async (): Promise<void> => {
 
 export function getPlaneMouseControls(_planeId: OrthoView): * {
   return {
-    leftDownMove: (delta: Point2, pos: Point2) => {
+    leftDownMove: (delta: Point2, pos: Point2, _id, event) => {
       const { tracing } = Store.getState();
       const volumeTracing = enforceVolumeTracing(tracing);
       const tool = getVolumeTool(volumeTracing);
       const contourTracingMode = getContourTracingMode(volumeTracing);
 
       if (tool === VolumeToolEnum.MOVE) {
         const state = Store.getState();
-        const { activeViewport } = state.viewModeData.plane;
-        const v = [-delta.x, -delta.y, 0];
-        Store.dispatch(movePlaneFlycamOrthoAction(v, activeViewport, true));
+        if (state.tracing.skeleton != null && event.ctrlKey) {
+          skeletonController.moveNode(delta.x, delta.y);
```

Comment:
Can we avoid that the volume tracing controller imports the skeleton controller? The `plane_controller` already switches between volume vs. skeleton controller depending on the current tool (see `createToolDependentHandler` usage in `plane_controller`). Therefore, I would have thought that the volume tracing planer controller doesn't have to know about moving nodes. Consequently, one couldn't move nodes when being in trace or brush mode, but this is perfectly fine.
Or is there another reason why this wouldn't work?

Reply:
I just resolved that. The code is cleaner now. Thanks for mentioning this :smile: